code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -369,20 +369,20 @@
 		return
 	to_chat(L, "<span class='notice'>This food has a funny taste!</span>")
 	
-/datum/reagent/rat_spit/overdose_start(mob/living/L)
+/datum/reagent/rat_spit/overdose_start(mob/living/carbon/C)
```

Comment:
Don't change the variable types or rename the variable names when overriding procs--it breaks some stuff like named arguments.

Instead, make this `mob/living/M`, then recast it to `var/mob/living/carbon/user = M` inside your code (after confirming it is a carbon).